### ISSUE TITLE
Deprecate docker-worker with generic-worker

### ISSUE THREAD

**Issue by mostlygeek** [taskcluster/taskcluster#5321]
(place holder for discussions)

**Comment by escapewindow** [comment]
> Questions:
> 
>     * who will be affected by deleting docker-worker ?

If `deleting docker-worker` includes deleting the AMIs, all automation in the FirefoxCI cluster will stop working, and Mozilla will not be able to ship releases.

If `deleting docker-worker` means deleting the code but not deleting the AMIs, we are still blocked from migrating to GCP from AWS unless we either set up automation to create the equivalent of AMIs in GCP, and we lose the ability to create security or other emergency fixes without creating a legacy branch. If we create a legacy branch, we're essentially not deleting the code, just moving it to another branch, which doesn't seem like any sort of win.

>     * what functionality gaps between docker-worker and generic-worker?

Since we've discussed various options to migrate before:

- If we add docker-worker schema support to generic-worker, then we probably need to add docker image downloading + verification (in-generic-worker code? a standalone helper tool?), and find the various other features that are in use that block a drop-in replacement;
- If we're thinking about moving the docker commands into `task.payload.command` I'm going to hard-veto that. We require Chain of Trust to verify the Chain of Custody of binaries that Mozilla Releng ships, and command lines are near impossible to securely verify at scale.

>     * what migration path is appropriate from docker-worker to generic-worker?
> 
>     * timelines and milestones for the work

We've had multiple discussions of all of this. For instance,

- [Docker worker deprecation work](https://docs.google.com/document/d/1Ljj4x-yCilK1a0TO09JA2vnSqa4au0apOOjHfhryg28/edit?skip_itp2_check=true&pli=1#)
- Row 6 in [Releng Priorities for Taskcluster](https://docs.google.com/spreadsheets/d/1lKOeg4IjFQpvMGveIfEL2pPF4Umz29scopvlJ64Vjms/edit#gid=0)
- [Releng Taskcluster Requests](https://docs.google.com/document/d/1Z5ViHdhTz3DqdvkHVw26i1mQ2dQCOEJqzbXYwNghzPI/edit)
- [Decision Brief: Support for docker-worker in Taskcluster](https://docs.google.com/document/d/1xC9axqIIblQ8mxyY2h0QJCTLGxgNeNBiTsOcAHIWv7Q/edit)

Most likely if we wanted to go this route, we should have done this work in 2020 or 2021, and we'd be unblocked for the GCP migration. We could still do it in 2022, but given our current velocity addressing worker-manager and generally relearning how Taskcluster works, I would be surprised if we can do this work without holding up the GCP migration.

**Comment by mostlygeek** [comment]
Does it make sense for generic-worker to be able to support (docker-worker's payload schema)[https://github.com/taskcluster/taskcluster/blob/main/workers/docker-worker/schemas/v1/payload.yml] and be a drop in replacement?

**Comment by escapewindow** [comment]
Yes, that's `5. Long term: Implement docker-worker compatibility layer in generic-worker (eta: 6mo - calendar time)` in the first link above.
